component/src/main/java/org/wso2/extension/siddhi/store/hbase/HBaseEventTable.java

```diff
@@ -258,18 +260,22 @@ protected CompiledExpression compileSetAttribute(ExpressionBuilder expressionBui
     @Override
     protected void connect() throws ConnectionUnavailableException {
         Configuration config = HBaseConfiguration.create();
-        storeAnnotation.getElements().stream()
-                .filter(Objects::nonNull)
-                .filter(element -> !(element.getKey().equals(ANNOTATION_ELEMENT_TABLE_NAME)
-                        || element.getKey().equals(ANNOTATION_ELEMENT_CF_NAME)))
-                .filter(element -> !(HBaseTableUtils.isEmpty(element.getKey())
-                        || HBaseTableUtils.isEmpty(element.getValue())))
-                .forEach(element -> config.set(element.getKey(), element.getValue()));
         try {
+            storeAnnotation.getElements().stream()
+                    .filter(Objects::nonNull)
+                    .filter(element -> !(element.getKey().equals(ANNOTATION_ELEMENT_TABLE_NAME)
+                            || element.getKey().equals(ANNOTATION_ELEMENT_CF_NAME)))
+                    .filter(element -> !(HBaseTableUtils.isEmpty(element.getKey())
+                            || HBaseTableUtils.isEmpty(element.getValue())))
+                    .forEach(element -> config.set(element.getKey(), element.getValue()));
+            HBaseAdmin.checkHBaseAvailable(config);
             this.connection = ConnectionFactory.createConnection(config);
         } catch (IOException e) {
-            throw new ConnectionUnavailableException("Failed to initialize store for table name '" +
-                    this.tableName + "': " + e.getMessage(), e);
+            throw new ConnectionUnavailableException("Failed to initialize store for table name: " + this.tableName +
+                    "': " + e.getMessage(), e);
+        } catch (ServiceException e) {
+            throw new ConnectionUnavailableException("Failed to initialize store for table name: " + this.tableName +
+                    "': " + e.getMessage(), e);
```

Comment:
I think we need to close the connection at the end of the try block with finally. 
Please validate and apply the changes.

Reply:
https://hbase.apache.org/apidocs/org/apache/hadoop/hbase/client/ConnectionFactory.html

Reply:
If something happened during the connect method, the disconnect will be called by siddhiManager. By disconnect, it will close the connection.